src/main/java/dev/socketmods/socketnukes/explosion/types/VanillaExplosionType.java

```diff
@@ -0,0 +1,244 @@
+                                yTemp = yTemp / avgDist;
+                                zTemp = zTemp / avgDist;
+                                float rayAngle = radius * (0.7F + worldIn.rand.nextFloat() * 0.6F);
+                                double sourceX = source.getX();
+                                double sourceY = source.getY();
+                                double sourceZ = source.getZ();
+
+                                for (; rayAngle > 0.0F; rayAngle -= 0.22500001F) {
+                                    BlockPos blockpos = new BlockPos(sourceX, sourceY, sourceZ);
+                                    BlockState blockstate = worldIn.getBlockState(blockpos);
+                                    FluidState fluidstate = worldIn.getFluidState(blockpos);
+
+                                    ExplosionContext dummyContext = new ExplosionContext();
+
+                                    Optional<Float> optional = dummyContext.getExplosionResistance(vanillaExplosion, worldIn, blockpos, blockstate, fluidstate);
+                                    if (optional.isPresent()) {
+                                        rayAngle -= (optional.get() + 0.3F) * 0.3F;
+                                    }
+
+                                    if (rayAngle > 0.0F) {
+                                        set.add(blockpos);
+                                    }
+
+                                    sourceX += xTemp * (double) 0.3F;
+                                    sourceY += yTemp * (double) 0.3F;
+                                    sourceZ += zTemp * (double) 0.3F;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                meta.affectedBlocks.addAll(set);
+                float radiusx2 = radius * 2.0F;
+                int eastBound = MathHelper.floor(source.getX() - (double) radiusx2 - 1.0D);
+                int westBound = MathHelper.floor(source.getX() + (double) radiusx2 + 1.0D);
+                int lowerBound = MathHelper.floor(source.getY() - (double) radiusx2 - 1.0D);
+                int upperBound = MathHelper.floor(source.getY() + (double) radiusx2 + 1.0D);
+                int southBound = MathHelper.floor(source.getZ() - (double) radiusx2 - 1.0D);
+                int northBound = MathHelper.floor(source.getZ() + (double) radiusx2 + 1.0D);
+
+                List<Entity> list = worldIn.getEntitiesWithinAABBExcludingEntity(placer, new AxisAlignedBB(eastBound, lowerBound, southBound, westBound, upperBound, northBound));
+
+                vanillaExplosion = new DummyExplosion(worldIn, placer, source.getX(), source.getY(), source.getZ(), radiusx2, meta.affectedBlocks);
+                ForgeEventFactory.onExplosionDetonate(worldIn, vanillaExplosion, list, radiusx2);
+                Vector3d explosionPos = new Vector3d(source.getX(), source.getY(), source.getZ());
+
+                for (Entity currentEntity : list) {
```

Comment:
Pull this block into `ExtendedExplosionType`? Its also used in the `CubicExplosionType`

Reply:
This removes versatility from the system. Note that the Null explosions do no logic. This is duplicated by intent.